src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -106,7 +122,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return _method.ToUpperInvariant().GetHashCode();
+            if (_hashcode == 0)
+            {
+                _hashcode = _method.ToUpperInvariant().GetHashCode();
```

Comment:
Other than for the assignment of the ```s_``` fields, is it very common for the known strings "GET", "POST", etc. to be used in constructing additional HttpMethod instances?  If so, I'm wondering whether it'd be useful to augment this to be something like:
```C#
if (_hashCode == 0)
{
    switch (_method)
    {
        case "GET": _hashcode = s_getMethod.GetHashCode(); break;
        case "POST": _hashCode = s_postMethod.GetHashCode(); break;
        ...
        default: _hashcode = _method.ToUpperInvariant().GetHashCode();
    }
}
```
or something like that.